aieng-ui/backend/app/runtime_registry/aieng.py

```
@@ -976,6 +976,36 @@ def _tool_aieng_apply_shape_ir_patch(inp: dict[str, Any], _ctx: dict[str, Any])
+                    triggering_tool="aieng.apply_shape_ir_patch",
+                    affected_artifacts=[
+                        "results/computed_metrics.json",
+                        "results/result_summary.json",
+                        "simulation/cae_mapping.json",
+                        "simulation/mesh/mesh_metadata.json",
+                        "analysis/field_regions.json",
+                        "cae/*",
+                    ],
+                )
+            except Exception:
+                log_exception(
```

Ruff (0.16.3)

[error] Undefined name `log_exception`

(F821)